### DISCUSSION TITLE
Handling minor and major updates and upgrades?

### DISCUSSION THREAD

**Discussion by mkastner** [keystonejs/keystone#8184] [Questions] [answered]
Is there any documentation on how to handle major and minor updates and upgrades?

Since keystone itself is installed globally, what are the proper steps to update and upgrade an keystone application?

I would assume (yeah, I know, never assume 'cause it makes an "a.." out of "u" and "me") first update the global keystone installation.

But then, what? Run an install or upgrade script over the existing application?

**Comment by flexdinesh** [comment, marked as answer]
All our [releases](https://github.com/keystonejs/keystone/releases) have notes on breaking changes (if any). Every breaking change has a link to the pull request with before vs after style changes that need to be done for a successful upgrade.

[Example release with breaking changes](https://github.com/keystonejs/keystone/releases/tag/2022-10-19)
[Example pull request with upgrade notes](https://github.com/keystonejs/keystone/pull/7943)

**Comment by mkastner** [comment]
@flexdinesh Most likely I wasn't sufficiently specific with my question. I am not so much concerned with what to upgrade, but with how.

Let's say, I have an app generated by a global keystone 6.x.x installation and there is a minor version upgrade to 6.x.y .

Do I need to run the (I'm using yarn) yarn global upgrade on the global keystone and then use the global installation to "regenerate" the app's dependencies or do I need to run a yarn upgrade on the app itself?

I apologize, if my questions don't make sense.

**Reply by flexdinesh** [reply]
No worries. We appreciate the question. It helps others who might have the same question.

You will need to manually upgrade each version depending on what your current version is and with the help of release notes. We don't have tools to automate Keystone version upgrades.
